web_src/js/components/DiffFileTree.vue

```diff
@@ -109,15 +108,17 @@ export default {
     updateVisibility(visible) {
       this.fileTreeIsVisible = visible;
       localStorage.setItem(LOCAL_STORAGE_KEY, this.fileTreeIsVisible);
-      this.adjustToggleButton(this.fileTreeIsVisible);
+      this.updateState(this.fileTreeIsVisible);
     },
-    adjustToggleButton(visible) {
-      const [toShow, toHide] = document.querySelectorAll('.diff-toggle-file-tree-button .icon');
-      toShow.classList.toggle('gt-hidden', visible);  // hide the toShow icon if the tree is visible
-      toHide.classList.toggle('gt-hidden', !visible); // similarly
-
-      const diffTree = document.getElementById('diff-file-tree');
-      diffTree.classList.toggle('gt-hidden', !visible);
+    updateState(visible) {
+      const btn = document.querySelector('.diff-toggle-file-tree-button');
+      const [toShow, toHide] = btn.querySelectorAll('.icon');
+      const tree = document.getElementById('diff-file-tree');
+      const newTooltip = btn.getAttribute(`data-${visible ? 'hide' : 'show'}-text`);
```

Comment:
To make the attribute searchable, how about:

```suggestion
      const newTooltip = btn.getAttribute(visible ? 'data-hide-text' : 'data-show-text');
```

I think we have done so when accessing `data-foo-bar`: avoid using `.dataset.fooBar` or `.data('foo-bar')`.

Reply:
Yeah we can do this for easier greppability. Also need to do the same change in JS.

Reply:
Done in https://github.com/go-gitea/gitea/pull/23553/commits/36a6926b726e18189562e7b1dc6cd7b30a8f16fb.